src/udx.c

```diff
@@ -457,7 +455,7 @@ clear_outgoing_packets (udx_stream_t *stream) {
       }
     }
 
-    if (pkt->type & UDX_PACKET_FREE_ON_SEND) {
+    if (pkt->type & (UDX_PACKET_STREAM_STATE | UDX_PACKET_STREAM_DESTROY)) {
```

Comment:
I think we can revert this now that `UDX_PACKET_FREE_ON_SEND` is in the shared header.